src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -0,0 +1,361 @@
+        if (securableObjectType == SecurableObjectType.OrganizationAtlasColumn) {
+            val organizationAtlasColumn = organizationExternalDatabaseColumns[securableObjectId]!!
+            tableName = organizationExternalDatabaseTables[organizationAtlasColumn.tableId]!!.name
+            columnName = organizationAtlasColumn.name
+        } else {
+            val organizationAtlasTable = organizationExternalDatabaseTables[securableObjectId]!!
+            tableName = organizationAtlasTable.name
+            columnName = ""
+        }
+        //add checks in here for map indexing
+        return Pair(tableName, columnName)
+    }
+
+    fun createOrganizationExternalDatabaseTable(orgId: UUID, table: OrganizationExternalDatabaseTable): UUID {
+        val principal = Principals.getCurrentUser()
+        Principals.ensureUser(principal)
+
+        val tableFQN = FullQualifiedName(orgId.toString(), table.name)
+        checkState(organizationExternalDatabaseTables.putIfAbsent(table.id, table) == null,
+                "OrganizationExternalDatabaseTable ${tableFQN.fullQualifiedNameAsString} already exists")
+        aclKeyReservations.reserveIdAndValidateType(table, tableFQN::getFullQualifiedNameAsString)
+
+        val tableAclKey = AclKey(orgId, table.id)
+        authorizationManager.setSecurableObjectType(tableAclKey, SecurableObjectType.OrganizationAtlasTable)
+        authorizationManager.addPermission(tableAclKey, principal, EnumSet.allOf(Permission::class.java))
+        //eventBus?
+
+        return table.id
+    }
+
+    fun createOrganizationExternalDatabaseColumn(orgId: UUID, column: OrganizationExternalDatabaseColumn): UUID {
+        val principal = Principals.getCurrentUser()
+        Principals.ensureUser(principal)
+
+        checkState(organizationExternalDatabaseTables[column.tableId] != null,
+                "OrganizationExternalDatabaseColumn ${column.name} belongs to a table that does not exist")
+        val columnFQN = FullQualifiedName(column.tableId.toString(), column.name)
+        checkState(organizationExternalDatabaseColumns.putIfAbsent(column.id, column) == null,
+                "OrganizationExternalDatabaseColumn ${columnFQN.fullQualifiedNameAsString} already exists")
+        aclKeyReservations.reserveIdAndValidateType(column, columnFQN::getFullQualifiedNameAsString)
+
+        val columnAclKey = AclKey(orgId, column.tableId, column.id)
+        authorizationManager.setSecurableObjectType(columnAclKey, SecurableObjectType.OrganizationAtlasColumn)
+        authorizationManager.addPermission(columnAclKey, principal, EnumSet.allOf(Permission::class.java))
+
+        return column.id
+    }
+
+    fun createNewOrganizationExternalDatabaseTable(orgId: UUID, tableName: String, columnNameToSqlType: LinkedHashMap<String, String>) {
+        //TODO figure out title stuff
+        val table = OrganizationExternalDatabaseTable(Optional.empty(), tableName, "title", Optional.empty(), orgId)
```

Comment:
What does "title" do?

Reply:
nothing useful currently. It's included in the OrganizationExternalDatabase object simply because it's a child class of AbstractSecurableObject. I'm not sure if it is necessary / if the title field can just serve as the name field of the column or table?